odbc_fdw.c

```diff
@@ -615,117 +602,109 @@ odbcGetTableSize(char *svr_dsn, char *svr_driver, char *svr_host, char *svr_port
 
-    /* Allocate a statement handle */
-    SQLAllocHandle(SQL_HANDLE_STMT, dbc, &stmt);
-
-    if (sql_count == NULL)
-    {
-        /* Get quote char */
-        getQuoteChar(dbc, &quote_char);
-
-        /* Get name qualifier char */
-        getNameQualifierChar(dbc, &name_qualifier_char);
-
-        initStringInfo(&sql_str);
-        appendStringInfo(&sql_str, "SELECT COUNT(*) FROM %s%s%s%s%s%s%s",
-                         quote_char.data, schema_name, quote_char.data,
-                         name_qualifier_char.data,
-                         quote_char.data, svr_table, quote_char.data);
-    }
-    else
-    {
-        initStringInfo(&sql_str);
-        appendStringInfo(&sql_str, "%s", sql_count);
-    }
+	/* Connect to the DSN */
+	ret = SQLDriverConnect(dbc, NULL, (SQLCHAR *) conn_str.data, SQL_NTS,
+                           OutConnStr, 1024, &OutConnStrLen, SQL_DRIVER_COMPLETE);
 
-#ifdef DEBUG
-    elog(NOTICE, "%s", sql_str.data);
-#endif
-    ret = SQLExecDirect(stmt, (SQLCHAR *) sql_str.data, SQL_NTS);
-    if (SQL_SUCCEEDED(ret))
-    {
-        SQLFetch(stmt);
-        /* retrieve column data as a big int */
-        ret = SQLGetData(stmt, 1, SQL_C_UBIGINT, &table_size, 0, &indicator);
-        if (SQL_SUCCEEDED(ret))
-        {
-            *size = (unsigned int) table_size;
-        }
-    }
-    else
-    {
-        elog(NOTICE, "Oops!");
-    }
-
-    /* Free handles, and disconnect */
-    if (stmt)
-    {
-        SQLFreeHandle(SQL_HANDLE_STMT, stmt);
-        stmt = NULL;
-    }
-    if (dbc)
-    {
-        SQLFreeHandle(SQL_HANDLE_DBC, dbc);
-        dbc = NULL;
-    }
-    if (env)
-    {
-        SQLFreeHandle(SQL_HANDLE_ENV, env);
-        env = NULL;
-    }
-    if (dbc)
-        SQLDisconnect(dbc);
-#ifdef DEBUG
-    elog(NOTICE, "Count:   %u", *size);
-#endif
+	#ifdef DEBUG
+		if (SQL_SUCCEEDED(ret))
+			elog(DEBUG1, "Successfully connected to driver");
+		else
+		{
+			extract_error("SQLDriverConnect", dbc, SQL_HANDLE_DBC);
+		}
+	#endif
+
+	/* Allocate a statement handle */
+	SQLAllocHandle(SQL_HANDLE_STMT, dbc, &stmt);
+
+	if (sql_count == NULL)
+	{
+		/* Get quote char */
+		getQuoteChar(dbc, &quote_char);
+
+		/* Get name qualifier char */
+		getNameQualifierChar(dbc, &name_qualifier_char);
+
+		initStringInfo(&sql_str);
+		appendStringInfo(&sql_str, "SELECT COUNT(*) FROM %s%s%s%s%s%s%s",
+		                 quote_char.data, schema_name, quote_char.data,
+		                 name_qualifier_char.data,
+		                 quote_char.data, svr_table, quote_char.data);
+	}
+	else
+	{
+		initStringInfo(&sql_str);
+		appendStringInfo(&sql_str, "%s", sql_count);
+	}
+
+	ret = SQLExecDirect(stmt, (SQLCHAR *) sql_str.data, SQL_NTS);
+	if (SQL_SUCCEEDED(ret))
+	{
+		SQLFetch(stmt);
+		/* retrieve column data as a big int */
+		ret = SQLGetData(stmt, 1, SQL_C_UBIGINT, &table_size, 0, &indicator);
+		if (SQL_SUCCEEDED(ret))
+		{
+			*size = (unsigned int) table_size;
+		}
+	}
+	else
+	{
+		elog(DEBUG1, "Oops!");
```

Comment:
In my short experience with the code, those "Oops!" actually signaled pretty damn serious problems (wrong parameters or such that made it not work at all).

I'd take note of this for the future, in order to avoid mixing up the style changes with anything else, to add a more informative trace than just "Oops!" and probably raise it to WARNING.